source/common/protobuf/utility.h

```diff
@@ -48,10 +49,26 @@ class RepeatedPtrUtil {
 class MessageUtil {
 public:
   static std::size_t hash(const Protobuf::Message& message) {
-    return std::hash<std::string>{}(message.SerializeAsString());
+    std::string text;
```

Comment:
Out of curiosity what was wrong with the old version? Maybe a small comment on what all of this is for?